src/hooks/useAudioDeviceSelection.ts

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useRef, useState } from "react";
+
+import { _t } from "../languageHandler";
+import MediaDeviceHandler, { MediaDeviceKindEnum } from "../MediaDeviceHandler";
+import { requestMediaPermissions } from "../utils/media/requestMediaPermissions";
+
+interface State {
+    devices: MediaDeviceInfo[];
+    device: MediaDeviceInfo | null;
+}
+
+export const useAudioDeviceTooltipSelection = (
```

Comment:
Not sure tooltip is still relevant in the name

Reply:
It isn't :/ I will update it